server/etcdserver/api/etcdhttp/health.go

```diff
@@ -193,3 +253,187 @@ func checkAPI(lg *zap.Logger, srv ServerHealth, serializable bool) Health {
 	lg.Debug("serving /health true")
 	return h
 }
+
+type HealthCheck func(ctx context.Context) error
+
+type CheckRegistry struct {
+	path           string
+	failCounter    prometheus.Counter
+	successCounter prometheus.Counter
+
+	checks map[string]HealthCheck
+	// the list of checkNames are needed to ensure the order of checks.
+	checkNames []string
+}
+
+func installLivezEndpoints(lg *zap.Logger, mux *http.ServeMux, server ServerHealth) {
+	reg := CheckRegistry{path: "/livez", checks: make(map[string]HealthCheck), failCounter: livezFailed, successCounter: livezSuccess}
+	reg.Register("serializable_read", serializableReadCheck(server))
+	reg.InstallHttpEndpoints(lg, mux)
+}
+
+func installReayzEndpoints(lg *zap.Logger, mux *http.ServeMux, server ServerHealth) {
+	reg := CheckRegistry{path: "/readyz", checks: make(map[string]HealthCheck), failCounter: readyzFailed, successCounter: readyzSuccess}
+	reg.Register("data_corruption", activeAlarmCheck(server, pb.AlarmType_CORRUPT))
+	reg.Register("serializable_read", serializableReadCheck(server))
+	reg.InstallHttpEndpoints(lg, mux)
+}
+
+func (reg *CheckRegistry) Register(name string, check HealthCheck) {
+	reg.checkNames = append(reg.checkNames, name)
+	reg.checks[name] = check
+}
+
+func (reg *CheckRegistry) InstallHttpEndpoints(lg *zap.Logger, mux *http.ServeMux) {
+	reg.installRootEndpoint(lg, mux)
+	for checkName, check := range reg.checks {
+		reg.installCheckEndpoint(checkName, check, mux)
+	}
+}
+
+func (reg *CheckRegistry) runHealthChecks(ctx context.Context, checkNames []string) (Health, error) {
```

Comment:
Don't understand why `runHealthChecks` cannot reused in individual check handler. 

Otherwise, there is no metric emitted for individual check.